firmware/src/adc.c

```diff
@@ -106,46 +87,40 @@ void adc_init(void) {
                 | (0 << COM0B1) | (0 << COM0B0);        // do nothing with OC0B
         TCCR0B  =   (0 << WGM02)                        // Timer 0 in Mode 2 = CTC (clear on compare)
                 | (0 << FOC0A) | (0 << FOC0B)           // dont force outputs
-                | (1 << CS02)                           // clock enabled, prescaller = 1024
-                | (0 << CS01)
-                | (1 << CS00);
+                | (0 << CS02)                           // clock enabled, prescaller = 64
+                | (1 << CS01)
+                | (0 << CS00);
 
-        OCR0A  =    20;                                 // Valor para igualdade de comparacao A para frequencia de ~1500 Hz
+        OCR0A  =    199;                                 // Valor para igualdade de comparacao A para frequencia de ~5000 Hz

```

Comment:
Please compute this in a function to be clear what is the frequency and be easy to change in the future